tests/ssl/mutual_authentication_test.py

```diff
@@ -0,0 +1,137 @@
+import os
+
+from unittest import skipIf
+from tests.base import HazelcastTestCase
+from hazelcast.client import HazelcastClient
+from hazelcast.config import PROTOCOL
+from hazelcast.exception import HazelcastError
+from tests.util import is_oss, get_ssl_config, configure_logging, get_abs_path
+
+
+@skipIf(is_oss(), "SSL/TLS is only supported with enterprise server.")
+class SSLAuthenticationTest(HazelcastTestCase):
+    current_directory = os.path.dirname(__file__)
+    rc = None
+    mutual_auth = True
+    ma_req_xml = get_abs_path(current_directory, "hazelcast-ma-required.xml")
+    ma_opt_xml = get_abs_path(current_directory, "hazelcast-ma-optional.xml")
+
+    @classmethod
+    def setUpClass(cls):
+        configure_logging()
+
+    def setUp(self):
+        self.rc = self.create_rc()
+
+    def tearDown(self):
+        self.rc.exit()
+
+    def test_required_client_and_server_authenticated(self):
+        cluster = self.create_cluster(self.rc, self.configure_cluster(True))
+        member = cluster.start_member()
+        client = HazelcastClient(get_ssl_config(True,
+                                                get_abs_path(self.current_directory, "server1-cert.pem"),
+                                                get_abs_path(self.current_directory, "client1-cert.pem"),
+                                                get_abs_path(self.current_directory, "client1-key.pem"),
+                                                protocol=PROTOCOL.TLSv1))
```

Comment:
default value of the protocol is protocol.TLS which is an alias for TLSv1.2. Since member is configured with TLSv1, for each test I specify it as TLSv1